### PR TITLE
Upgrade material-mkdocs to major version 8 

### LINKED ISSUE #4491: Update website to new major version v8 of material-mkdocs
We have being waiting to get support for [code annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations)
image

It looks like it would be included in [v8](https://github.com/squidfunk/mkdocs-material/releases/tag/8.0.0b1), currently, we use v7 we need to look at the [breaking changes](https://github.com/squidfunk/mkdocs-material/issues/3219)

/kind mkdocs

### PR DESCRIPTION
**Pull request by csantanapr**
Fixes #4491

Upgrade guide is here https://squidfunk.github.io/mkdocs-material/upgrade/#whats-new

### CODE REVIEW

File: docs/getting-started/first-service.md [hunk 1]
```diff
@@ -14,11 +14,10 @@ Since our "Hello world" Service is being deployed as a Knative Service, not a Ku
     --image gcr.io/knative-samples/helloworld-go \
     --port 8080 \
     --env TARGET=World \
-    --revision-name=world
+    --revision-name=world # (1)
```
Comment (snneji) [inline docs/getting-started/first-service.md:17]: I found these annotations a bit hard to spot. I'm wondering whether we can make them easier to see - like in the Red Hat docs:
Screen Shot 2022-02-07 at 12 48 53

Comment (snneji) [reply docs/getting-started/first-service.md:17]: Also, are we able to get numbering for the annotations?

Comment (psschwei) [reply docs/getting-started/first-service.md:17]: looks like numbers are possible: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/?h=code+annot#annotations-with-numbers

Don't see anything yet about how to change their appearance

**Comment by psschwei** [comment]
One thing to note: with code annotations, I no longer get the nice "copied to clipboard" message when clicking that button

**Comment by csantanapr** [comment]
> One thing to note: with code annotations, I no longer get the nice "copied to clipboard" message when clicking that button

Good find @psschwei 

It looks like this is unrelated to code annotations, is related to the new version of mkdocs 
I will open an issue in mkdocs to find out more

**Comment by csantanapr** [comment]
/hold

@psschwei The "copied to clipboard" shows find in main branch, but it doesn't look fine in older version. We need to update the `mkdocs.yaml` in the old versions but I want to test it first.

**Comment by csantanapr** [comment]
Going to remove code annotations from this PR, and just deal with upgrade mkdocs. Then we can look at code annotations to see if we want to added them or not.

**Comment by csantanapr** [comment]
@psschwei checkout the preview, in main (pre-release) it shows the icon and the message at the bottom
https://620165046b0d77000761aed8--knative.netlify.app/development/install/yaml-install/serving/install-serving-with-yaml/#install-a-networking-layer
image

On version 1.2 there is no icon, I'm guessing we need to update the mkdocs.yaml in the other branches that's my current guess.

https://620165046b0d77000761aed8--knative.netlify.app/docs/install/serving/install-serving-with-yaml/
image

**Comment by psschwei** [comment]
That's odd -- it works fine on both the v1.2 and pre-release branches on knative.dev.

**Comment by csantanapr** [comment]
> That's odd -- it works fine on both the v1.2 and pre-release branches on knative.dev.

Wait what? from the preview URL? https://620165046b0d77000761aed8--knative.netlify.app/docs/install/serving/install-serving-with-yaml/
Can you try this? I tried on incognito mode
image

**Comment by psschwei** [comment]
> @psschwei Wait what? from the preview URL? 

No, from the actual website ([https://knative.dev](https://knative.dev)). On the preview I see the same behavior you see. Given that this PR isn't touching the 1.2 branch, it's strange that the preview for it doesn't work. I would expect the v1.2 view to be same on both the preview and knative.dev

**Comment by csantanapr** [comment]
@psschwei I got it working, we need to touch `mkdocs.yaml` on every branch (1.2,1.1,1.0,0.26) and the update `main`

We need this update
```
- pymdownx.tabbed:
      alternate_style: true
```

I tested using my fork and building the whole site locally, here is 0.26 with the copy icon working
image

I will create the PRs for the branches.

**Comment by csantanapr** [comment]
@psschwei 
We need to merge these first:
https://github.com/knative/docs/pull/4713
https://github.com/knative/docs/pull/4712
https://github.com/knative/docs/pull/4711
https://github.com/knative/docs/pull/4710

**Comment by snneji** [comment]
Did you change the colour of the banner to black as part of the upgrade? 

The banner is yellow on https://knative.dev/v1.1-docs/:

Screen Shot 2022-02-08 at 10 21 17

But it's black in the preview https://deploy-preview-4705--knative.netlify.app/v1.1-docs/:

Screen Shot 2022-02-08 at 10 22 14


Also, I noticed a black strip above the header section at the top of the page when on the pre-release and latest versions. I'm not sure if it is just me though:

Screen Shot 2022-02-08 at 10 27 59

**Comment by psschwei** [comment]
> @psschwei We need to merge these first: #4713 #4712 #4711 #4710

Should we also need to bump the version of mkdocs in the release branches? Don't think it's needed to build the website proper (as best I can tell, requirements is installed from main), but in the off chance someone tries to build a specific branch from scratch might not be a bad idea.

**Comment by csantanapr** [comment]
@psschwei I did a full build with the changes on all branches. Here is the banner fixed

image

image

image

image

image

I think we are ready to `/lgtm` on all PRs including this one.

I will remove the hold on the old branches, and then last one this PR
https://github.com/knative/docs/pull/4713
https://github.com/knative/docs/pull/4712
https://github.com/knative/docs/pull/4711
https://github.com/knative/docs/pull/4710

**Comment by csantanapr** [comment]
You will not be able to see the yellow banner, and the fix for 1.2 black extra bar.
I had to do a full local build

**Comment by csantanapr** [comment]
@psschwei could also lgtm this one, that way I can do them all at once

**Pull request merged**